lib/Importer/ProtobufLoader.cpp

```diff
@@ -101,16 +100,39 @@ bool ProtobufLoader::hasNodeByName(llvm::StringRef name) const {
 }
 
 ProtobufLoader::ProtobufLoader(llvm::ArrayRef<const char *> tensorNames,
-                               llvm::ArrayRef<TypeRef> types, Function &F)
+                               llvm::ArrayRef<TypeRef> types, Function &F,
+                               llvm::Error *errPtr)
     : G_(F) {
   // Verify that the version of the library that we linked against is
   // compatible with the version of the headers we compiled against.
   GOOGLE_PROTOBUF_VERIFY_VERSION;
 
-  assert(tensorNames.size() == types.size() && "Invalid initialization list");
-  for (unsigned i = 0; i < tensorNames.size(); i++) {
-    assert(!hasNodeByName(tensorNames[i]) && "Input names have duplicate");
-    TEMP_EXIT_ON_ERR(createAndRegisterPlaceholder(tensorNames[i], types[i]));
+  // if errPtr already contains an error then don't continue with constructor
+  if (errPtr && *errPtr) {
+    return;
+  }
+
+  // Lambda to setup the ProtobufLoader and return any llvm::Errors that were
+  // raised.
+  auto setup = [&]() -> llvm::Error {
+    RETURN_ERR_IF_NOT(tensorNames.size() == types.size(),
+                      "Invalid initialization list");
+    for (unsigned i = 0; i < tensorNames.size(); i++) {
```

Comment:
LLVM typically recommend to do this kind of loop
https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop